fedimint-client/src/lib.rs

```diff
@@ -1494,19 +1506,12 @@ async fn get_config(
     db: &Database,
     config_source: Option<ConfigSource>,
 ) -> anyhow::Result<ClientConfig> {
-    let mut dbtx = db.begin_transaction().await;
-    let config_res = match dbtx
-        .find_by_prefix(&ClientConfigKeyPrefix)
-        .await
-        .next()
-        .await
-    {
-        Some((_, config)) => {
-            // TODO: Enable after <https://github.com/fedimint/fedimint/pull/2855>
-            // assert!(
-            //     config_source.is_none(),
-            //     "Alternative config source provided but config was found in DB"
-            // );
+    let config_res = match get_config_from_db(db).await {
```

Comment:
Where is this function used? Does the gateway ever use `get_config`?

Reply:
The static methods `get_config` and `get_config_from_db` are used in `ClientBuilder`. Gateway composes the client builder. See `build_stopped` method of the client builder

Reply:
gateway client builder now uses `get_config_from_db` to detect client rebuild scenario